advisor-backend/routes/schedule.js

```diff
@@ -208,6 +292,47 @@ const generateSchedule = async (
   return years;
 };
 
+const moveCourseToDifferentYear = (
+  schedule,
+  desiredYear,
+  originalYear,
+  course
+) => {
+  for (let i = 0; i < schedule.length; i++) {
+    const year = schedule[i];
+    // delete it from this year
+    if (year.number === originalYear) {
+      // for both semesters
+      year.semesters[0].classes = year.semesters[0].classes.filter(
+        (classItem) => classItem.id !== course.id
+      );
+      year.semesters[1].classes = year.semesters[0].classes.filter(
+        (classItem) => classItem.id !== course.id
+      );
+    }
+
+    // add it to this year
+    if (year.number === desiredYear) {
+      year.semesters[0].classes.push(course);
+      year.semesters[1].classes.push(course);
+    }
+  }
+};
+
+const reverseAdjList = (adjList) => {
+  let reversedAdjList = {};
+  for (const node in adjList) {
+    reversedAdjList[node] = [];
+  }
+
+  for (const node in adjList) {
+    for (const neighbor of adjList[node]) {
+      reversedAdjList[neighbor].push(node);
+    }
+  }
+  return reversedAdjList;
+};
+
```

Comment:
I don't understand what this is doing - from the function name it looks like it's reversing the order of something, but `adjList` is an object, so it's unordered. 

Are you trying to do like an inversion of keys and values?